src/main/java/net/neoforged/neoforge/items/SlotItemHandlerImmutable.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.items;
+
+import net.minecraft.world.item.ItemStack;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Slot class that can be used with immutable itemstack storages (Ex: {@link ComponentItemHandler}
+ */
+public class SlotItemHandlerImmutable extends SlotItemHandler {
```

Comment:
I think that a "hacky slot" (with a better name) is a useful thing to have even independently of item handlers. So I'd keep it separate :)

Reply:
Any suggestions for a name? I would assume something with `Immutable` maybe?

Also do you want an implementation that implements hacky slot and then copies a lot of functionality from `SlotItemHandler`?

Reply:
`StackCopySlot` or something? I fear that `Immutable` would suggest immutability of the slot itself. And maybe `getRealStack/setRealStack` -> `getStackCopy/setStackCopy`?

It might be possible to wrap a `SlotItemHandler`. I don't know if that would be a good idea.